Narrative bugfix

AdamZarger opened the pull request:
The `has_narrative` parameter got left out of aggregation filters when the filtering was refactored (due to `has_narrative` having some special logic for converting strings to bools). This PR adds back `has_narrative` as a filter option for aggregations.

flacoman91 commented:
this is what i see
![narrative](https://user-images.githubusercontent.com/1038306/87436956-b4237400-c5bb-11ea-87a9-085129557edf.gif)

passes for me.  i'll defer to others though as far as the code

JeffreyMFarley commented:
What about this?  From `es_builders.py`

```
    _OPTIONAL_FILTERS_STRING_TO_BOOL = ("has_narrative",)
```

Should this go away?  When did this stop working?

AdamZarger commented:
Good catch, thanks! That went away when we refactored the filters; the new way designed for V2 iis more robust so we were able to get rid of the custom logic previously needed. I have removed this old value.
